### PR TITLE
Capacitor 6

### PR DESCRIPTION
**Pull request by davidofwatkins**
This rebases our forked changes onto the `6.x` branch to support Capacitor 6. I chose to rebase/cherry-pick our changes rather than merge so that all our changes are within the same set of commits and it's easier to see what we've actually changed (and to avoid unrelated merge conflicts). I took the liberty to squash a few related commits that were made outside of Pull Requests.

> [!IMPORTANT]
> To be clear, there should be no new changes in this PR that weren't already authored by Watch Duty in `main` or by the Ionic team upstream.

### Branching Strategy

With this change, I'm proposing changes to the way that we manage this fork. Before, we would merge the latest upstream stable branch (previously `5.x`) into our `main` branch and release from there. This resulted in several unrelated merge conflicts when attempting to merge `6.x` into `main`. Instead, I propose we follow the branching strategy of the upstream Ionic repo:

- Our `5.x` branch contains our changes on top of the upstream `5.x` branch. I've already cherry-picked our commits onto this branch, and there's no diff with `main`.
- Our `6.x` branch (with this PR) will similarly contain our changes on top of the upstream `6.x` branch. When new versions of Capacitor 6 are released, we'll `git merge` the upstream `6.x` branch into this.
- `main` will be hard reset against the upstream `main` branch to remove any difference between the two initially. If/when we decide to formalize our internal changes into PRs that Capacitor might accept, we can branch off `main` and submit changes from feature branches.
    - Just to be safe, I've already created a backup of `main` called `main-bkp-cap-5`. We probably won't need to keep this long-term, especially since `main` has no diff against `5.x` currently.
- When Capacitor 7 comes out, we'll cherry-pick our changes from the `6.x` branch onto the `7.x` branch.

Please let me know how this sounds. Hopefully, we will shrink the delta between our fork and upstream over time!

### CODE REVIEW

File: camera/rollup.config.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ export default {
     {
       file: 'dist/plugin.js',
       format: 'iife',
-      name: 'capacitorCamera',
+      name: 'WatchdutyCamera',
```
Comment (gabeschine) [inline camera/rollup.config.js]: Tiniest of nits: the diff changes from lower-camel to upper-camel case. Is that intentional?

**Review by gabeschine — APPROVED** [review]
Yeah this sounds good.

Whatever strategy we choose, the risk seems somewhat low: we have a small number of changes to track here, and we don't expect much if any more.

File: camera/rollup.config.js (hunk 1 above)
Comment (davidofwatkins) [reply camera/rollup.config.js]: I'm not sure since it's been like this, but I agree. Updated.

**Review by davidemerritt — APPROVED** [review]
lgtm - while this is fresh in your mind can you write up the ticket for getting these added back into the camera plugin with the functional changes we added.

This would be a great volunteer project if we can get the ticket with clear requirements.

**Comment by davidofwatkins** [comment]
> lgtm - while this is fresh in your mind can you write up the ticket for getting these added back into the camera plugin with the functional changes we added.
> 
> This would be a great volunteer project if we can get the ticket with clear requirements.

Done: https://trello.com/c/JQM82ajM#comment-6718ad13511c09947067167b

**Pull request merged**